internal/runnerinstall/ecs.go

```diff
@@ -330,14 +345,76 @@ func (i *ECSRunnerInstaller) Uninstall(ctx context.Context, opts *InstallOpts) e
 			Services: []*string{service.ServiceArn},
 		})
 		if err != nil {
-			log.Debug("error waint for runner to deactivate: %s", err)
+			log.Debug("error waiting for runner to deactivate: %s", err)
 			ui.Output("Error waiting for Runner service to deactivate: %s", clierrors.Humanize(err), terminal.WithErrorStyle())
 			return err
 		}
 
 		s.Update("Runner uninstalled")
 	}
 	s.Done()
+
+	// TODO: Still attempt to delete the EFS volume if the ECS service
+	// uninstall fails
+	efsSvc := efs.New(sess)
+	fileSystemsResp, err := efsSvc.DescribeFileSystems(&efs.DescribeFileSystemsInput{
+		CreationToken: nil,
+		FileSystemId:  nil,
+		Marker:        nil,
+		MaxItems:      nil,
+	})
+	if err != nil {
+		return err
+	}
+	for _, fileSystem := range fileSystemsResp.FileSystems {
+		// Check if tags match ID, if so then delete things
+		for _, tag := range fileSystem.Tags {
+			if *tag.Key == "runner-id" && *tag.Value == opts.Id {
+				describeAccessPointsResp, err := efsSvc.DescribeAccessPoints(&efs.DescribeAccessPointsInput{
+					FileSystemId: fileSystem.FileSystemId,
+				})
+				if err != nil {
+					return err
+				}
+				for _, accessPoint := range describeAccessPointsResp.AccessPoints {
+					_, err = efsSvc.DeleteAccessPoint(&efs.DeleteAccessPointInput{AccessPointId: accessPoint.AccessPointId})
```

Comment:
As I understand it, it is required to delete the access point prior to the mount points.

The purpose for deleting the file system for the runner and not the server is that the server file system contains the server database. If we delete that via the CLI command, then having a chance to recover the database becomes much more difficult. The file system for the runner contains only a token and the runner ID, so it can be deleted with a low risk to impacting the user as they more likely won't need to "recover" a token or runner ID.

Reply:
I'm not sure I follow; when we uninstall the server, we do in fact delete the file system, correct? I'm just pointing out the differences here and the server uninstall path. If this code path works correctly and everything is cleaned up then I suppose there's no issue.